src/e2e/app/repo-tests/src/test_flight_sql.rs

```diff
@@ -0,0 +1,122 @@
+use kamu_node_e2e_common::KamuFlightSQLClient;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub async fn test_flight_sql_self_test(mut kamu_flight_sql_client: KamuFlightSQLClient) {
+    kamu_flight_sql_client.set_anonymous().await;
+
+    kamu_flight_sql_client
+        .flight_sql_assert_call(
+            "select 1 as value",
+            Some(indoc!(
+                "
+        message arrow_schema {
+          REQUIRED INT64 value;
+        }
+        "
+            )),
+            None,
+        )
+        .await;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub async fn test_flight_sql_anonymous(mut kamu_flight_sql_client: KamuFlightSQLClient) {
+    kamu_flight_sql_client
+        .kamu_api_client()
+        .auth()
+        .login_as_kamu()
+        .await;
+    kamu_flight_sql_client
+        .kamu_api_client()
+        .dataset()
+        .create_player_scores_dataset_with_data()
+        .await;
+
+    kamu_flight_sql_client.set_anonymous().await;
+
+    kamu_flight_sql_client
+        .flight_sql_assert_call(
+            "select offset, op, match_time, match_id, player_id, score from 'kamu/player-scores'",
+            Some(indoc!(
+                "
+        message arrow_schema {
+          REQUIRED INT64 offset;
+          REQUIRED INT32 op;
+          OPTIONAL INT64 match_time (TIMESTAMP(MILLIS,true));
+          OPTIONAL INT64 match_id;
+          OPTIONAL BYTE_ARRAY player_id (STRING);
+          OPTIONAL INT64 score;
+        }
+        "
+            )),
+            Some(indoc!(
+                "
+                +--------+----+----------------------+----------+-----------+-------+
+                | offset | op | match_time           | match_id | player_id | score |
+                +--------+----+----------------------+----------+-----------+-------+
+                | 0      | 0  | 2000-01-01T00:00:00Z | 1        | Alice     | 100   |
+                | 1      | 0  | 2000-01-01T00:00:00Z | 1        | Bob       | 80    |
+                +--------+----+----------------------+----------+-----------+-------+
+                "
+            )),
+        )
+        .await;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub async fn test_flight_sql(mut kamu_flight_sql_client: KamuFlightSQLClient) {
+    let token = kamu_flight_sql_client
+        .kamu_api_client()
+        .auth()
+        .login_as_kamu()
+        .await;
+    kamu_flight_sql_client
+        .kamu_api_client()
+        .dataset()
+        .create_player_scores_dataset_with_data()
+        .await;
+
+    kamu_flight_sql_client.set_token(token);
+
+    kamu_flight_sql_client
+        .flight_sql_assert_call(
+            "select offset, op, match_time, match_id, player_id, score from 'kamu/player-scores'",
```

Comment:
`offset` is an unstable column -- in order to make the tests stable, I propose to exclude it from the query